astropy/utils/tests/test_data.py

```diff
@@ -1154,23 +1154,37 @@ def test_compressed_stream():
     gzipped_data = base64.b64decode(gzipped_data)
     assert isinstance(gzipped_data, bytes)
 
-    class FakeStream:
+    class FakeStream(io.RawIOBase):
         """
         A fake stream that has `read`, but no `seek`.
         """
 
         def __init__(self, data):
             self.data = data
 
-        def read(self, nbytes=None):
-            if nbytes is None:
+        def read(self, size=-1):
+            if size < 0:
                 result = self.data
                 self.data = b""
             else:
-                result = self.data[:nbytes]
-                self.data = self.data[nbytes:]
+                result = self.data[:size]
+                self.data = self.data[size:]
             return result
 
+        def readable(self):
+            return True
+
+        def readinto(self, b):
+            result = self.read(len(b))
+            b[:len(result)] = result
+            return len(result)
+
+        def seek(self, offset, whence=io.SEEK_SET):
+            raise OSError
+
+        def seekable(self):
+            return False
```

Comment:
The test seems quite different from the initial one, which was a simple object with just a `read` method. What's the goal here ?

Reply:
The goal of the changes here was to ensure that `FakeStream` would be a subclass of `io.RawIOBase`.